js/libs/form.js

```diff
@@ -854,7 +854,9 @@ Fliplet.Widget.instance('form-builder', function(data) {
                   field.value = value;
                   debouncedUpdate();
 
-                  if (hasChanged) {
+                  // To set correct date in the date picker
+                  // We want to trigger change on date field every time.
+                  if (hasChanged || !field._type === 'flDate') {
```

Comment:
You sure about that exclamation mark? I've never seen such block condition statement.

Reply:
@tonytlwu As I understand the work of the date picker in the date component correct, we should trigger a change every time we do a set() to tell the date picker set a correct date.
Should I remove this comment?

Reply:
@YaroslavOvdii I don't think @squallstar's question has anything to do with requirements. Your syntax looks wrong and possibly untested.

Reply:
@YaroslavOvdii have you tested this? @tonytlwu yes that's what I was referring to, I thought my comment was explicit enough but it probably wasn't.

@YaroslavOvdii to confirm, `!field._type` would produce `false`, and `false` is never going to equal `flDate`.

Reply:
@squallstar Ou I get it now, sorry for the misunderstanding.
Yes, I have tested it but it would be better to change it to `field._type === date`

Reply:
@YaroslavOvdii you sure you tested it and it works? From this fiddle it doesn't look like: https://jsfiddle.net/qkdto6L7/

Reply:
@squallstar Yes, you are right about that, it worked for me because I didn't test it correctly. I've tested it with connection with the LFD in the chain LFD -> edit item -> look a form builder. In that case `field.set()` will be working correctly.
If we use `field._type === 'flDate'` it will be working correctly for all other cases. Sorry for the long answer.